client/trino-cli/src/main/java/io/trino/cli/Pager.java

```diff
@@ -160,9 +209,38 @@ public static Pager create(List<String> command)
         }
     }
 
+    private static Pager createBuiltinPager(Terminal terminal)
+    {
+        Options opt = Options.compile(Less.usage()).parse(Arrays.asList("--quit-if-one-screen", "--no-init", "--chop-long-lines"));
+        Less less = new Less(terminal, Paths.get("").toAbsolutePath(), opt);
+        PipedInputStream in = new PipedInputStream();
+        final PipedOutputStream out;
+        try {
+            out = new PipedOutputStream(in);
+        }
+        catch (IOException ex) {
+            throw new RuntimeException(ex);
+        }
+        // TODO it might not be the best idea to run it in a different thread, since it's using the same terminal
+        // maybe try to wrap the client handler in a thread and block on less.run()
```

Comment:
Do we need this TODO be addressed?

Reply:
I was hoping to get some opinions about that during the review

Reply:
i don't know much about java and terminal(s). 

cc @electrum